src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherPrefetchIndices.template

```diff
@@ -0,0 +1,347 @@
+
+            {Op1VectorType}<{Op1BaseType}> falseMask = Sve.CreateFalseMask{Op1BaseType}();
+
+            {Isa}.{Method}(
+                falseMask,
+                (byte*)_dataTable.inArray2Ptr,
+                Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr),
+                {ValidPrefetch}
+            );
+        }
+
+        public void RunBasicScenario_NonFaulting()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_NonFaulting));
+
+            {Op1VectorType}<{Op1BaseType}> falseMask = Sve.CreateFalseMask{Op1BaseType}();
+
+            try
+            {
+                {Isa}.{Method}(
+                    falseMask,
+                    default,
+                    Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr),
+                    {ValidPrefetch}
+                );
+            }
+            catch
+            {
+                Succeeded = false;
+            }
+        }
+
+        public void RunBasicScenario_Invalid()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_Invalid));
+
+            try
+            {
+                {Isa}.{Method}(
+                    Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr),
+                    (byte*)_dataTable.inArray2Ptr,
+                    Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr),
+                    {InvalidPrefetch}
+                    );
+                Succeeded = false;
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+            }
+        }
+
+        public void RunReflectionScenario_UnsafeRead()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunReflectionScenario_UnsafeRead));
+
+            typeof({Isa}).GetMethod(nameof({Isa}.{Method}), new Type[] { typeof({Op1VectorType}<{Op1BaseType}>), typeof({Op2BaseType}*), typeof({Op3VectorType}<{Op3BaseType}>), typeof({Op4BaseType}) })
+                                     .Invoke(null, new object[] {
+                                        Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr),
+                                        Pointer.Box(_dataTable.inArray2Ptr, typeof(byte*)),
+                                        Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr),
+                                        {ValidPrefetch}
+                                     });
+        }
+
+        public void RunLclVarScenario_UnsafeRead()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunLclVarScenario_UnsafeRead));
+
+            var op1 = Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr);
+            var op2 = (byte*)_dataTable.inArray2Ptr;
+            var op3 = Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr);
+            {Isa}.{Method}(op1, op2, op3, {ValidPrefetch});
```

Comment:
@a74nh - while testing something else, I realized that we are not validating the result in these tests.